bufferserver/src/main/java/com/datatorrent/bufferserver/internal/LogicalNode.java

```diff
@@ -115,9 +116,10 @@ public void addConnection(WriteOnlyClient connection)
    */
   public void removeChannel(WriteOnlyClient client)
   {
-    for (PhysicalNode pn : physicalNodes) {
-      if (pn.getClient() == client) {
-        physicalNodes.remove(pn);
```

Comment:
I don't see what is wrong with the existing way to remove a channel other than an extra hash lookup. With Java 8 upgrade, it should be changed to 'physicalNodes.removeIf(pn -> pn.getClient().equals(client));'

Reply:
It's modifying the set directly as it is iterating through it without using the iterator causing the concurrent modification exception.

Reply:
Using Java 8 lambdas

Reply:
Please fix the build.